src/Query/Builder.php

```diff
@@ -789,24 +789,18 @@ public function increment($column, $amount = 1, array $extra = [], array $option
 
     public function incrementEach(array $columns, array $extra = [], array $options = [])
     {
-        $query = ['$inc' => $columns];
+        $stage['$addFields'] = $extra;
 
-        if ($extra) {
-            $query['$set'] = $extra;
-        }
-
-        // Protect
+        // Not using $inc for each column, because it would fail if one column is null.
         foreach ($columns as $column => $amount) {
-            $this->where(function ($query) use ($column) {
-                $query->where($column, 'exists', false);
-
-                $query->orWhereNotNull($column);
-            });
+            $stage['$addFields'][$column] = [
+                '$add' => [['$ifNull' => ['$' . $column, 0]], $amount],
```

Comment:
According to the [docs](https://www.mongodb.com/docs/manual/reference/operator/aggregation/add/), there's an optimisation we can leverage in MongoDB 6.1 and later:
> Starting in MongoDB 6.1 you can optimize the `$add` operation. To improve performance, group references at the end of the argument list.

```suggestion
                '$add' => [$amount, ['$ifNull' => ['$' . $column, 0]]],
```

Reply:
Thanks. It's strange that the server isn't able to apply this kind of optimisation itself.